File: problems/1299/solution.go
package main
func replaceElements(arr []int) []int {
    cm := -1
    
    for i := len(arr) - 1; i >= 0; i-- {
        temp := arr[i]
        
        arr[i] = cm
        
        cm = max(cm, temp)
    }
    
    return arr
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}

File: problems/1299/solution.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        m = arr[-1]
        arr[-1] = -1
        i = len(arr) - 2
        
        while i >= 0:
            cm = arr[i]
            arr[i] = m
            m = max(m, cm)
            i -= 1
            
        return arr